Mathlib/CategoryTheory/Galois/Basic.lean

```diff
@@ -123,8 +126,97 @@ noncomputable instance : ReflectsColimitsOfShape (Discrete PEmpty.{1}) F :=
 noncomputable instance : PreservesFiniteLimits F :=
   preservesFiniteLimitsOfPreservesTerminalAndPullbacks F
 
+instance : PreservesMonomorphisms F :=
+  preservesMonomorphisms_of_preservesLimitsOfShape F
+
+/- Fibre functors reflect monomorphisms. -/
+instance : ReflectsMonomorphisms F := ReflectsMonomorphisms.mk <| by
+  intro X Y f _
+  have : IsIso (pullback.fst : pullback (F.map f) (F.map f) ⟶ F.obj X) :=
+    fst_iso_of_mono_eq (F.map f)
+  have : IsIso (F.map (pullback.fst : pullback f f ⟶ X)) := by
+    rw [← PreservesPullback.iso_hom_fst]
+    exact IsIso.comp_isIso
+  have : IsIso (pullback.fst : pullback f f ⟶ X) := isIso_of_reflects_iso pullback.fst F
+  exact (pullback.diagonal_isKernelPair f).mono_of_isIso_fst
+
+/- Fibre functors are faithful. -/
+instance : Faithful F where
+  map_injective {X Y} := by
+    intro f g h
+    have : IsIso (equalizer.ι (F.map f) (F.map g)) := equalizer.ι_of_eq h
+    have : IsIso (F.map (equalizer.ι f g)) := by
+      rw [← equalizerComparison_comp_π f g F]
+      exact IsIso.comp_isIso
+    have : IsIso (equalizer.ι f g) := isIso_of_reflects_iso _ F
+    exact eq_of_epi_equalizer
+
 end FibreFunctor
 
+variable (F : C ⥤ FintypeCat.{w}) [FibreFunctor F]
+
+private lemma IsInitial.isInitial_iff_obj {D : Type v₁} [Category.{v₂} D] (G : C ⥤ D)
+    [PreservesColimitsOfShape (Discrete PEmpty.{1}) G]
+    [ReflectsColimitsOfShape (Discrete PEmpty.{1}) G] (X : C) :
+    Nonempty (IsInitial X) ↔ Nonempty (IsInitial (G.obj X)) := by
+  constructor
+  · intro ⟨h⟩
+    exact ⟨IsInitial.isInitialObj G X h⟩
+  · intro ⟨h⟩
+    exact ⟨IsInitial.isInitialOfObj G X h⟩
+
+private lemma Types.initial_iff_empty (X : Type w) : Nonempty (IsInitial X) ↔ IsEmpty X := by
```

Comment:
Why are you making these lemmas private? I could imagine others needing them. Probably they would need to go in other files?

Reply:
I moved the one for `Type u` in the corresponding file. The one for `FintypeCat` I am not sure about: There is no file `CategoryTheory.Limits.Shapes.FintypeCat` yet and I think I once asked @joelriou a similar question and they said, they prefer not introducing such a file and instead formulating statements like this for `ConcreteCategory`. In this case I am unsure if this is worth the generality? This holds for full subcategories of `Type u` that have the empty set, so I might as well say it holds for `FintypeCat`.

Reply:
I have now put some API on initials and terminals in concrete categories. Is this what you head in mind @adamtopaz ? Maybe we can somehow make the `forget FintypeCat` definition reducible, so that it automatically infers the instances from `FintypeCat.incl`? See the additional `haveI`s in the new version.